### ISSUE TITLE
Can Sprig preserve the amount of entries loaded from the Load More module on clicking back history button?

### ISSUE THREAD

**Issue by cannon303** [putyourlightson/craft-sprig#195]
### Question
Hi I am using the Load More module and it's working really great with a masonry style layout. The page is an archive of entries and it uses javascript to lay out entries in rows rather than the css default of columns. The entries are in date order and I have 10 years of entries (around 180) and they stack up in batches of 20 each time the user clicks the load more button.

My code is a bit more complex than this but essentially its using this functionality:

```
{{ sprig('_components/load-more') }}


{#--- _components/load-more.twig ---#}

{% set entries = craft.entries.offset(offset).limit(_limit).all() %}

{% for entry in entries %}
  {{ entry.title }}
  {{ entry.url }}
{% endfor %}

{% if entries %}
  <button sprig s-val:offset="{{ offset + _limit }}" s-target="this" s-swap="outerHTML">Load more</button>
{% endif %}
```

The problem I'm trying to overcome is when a user clicks one of the posts and then decides to click the history back button to get back to the archive they then have to click the load more button multiple times again to get back to their original place in the archive.

I've tried using Sprig's pagination functionality but that doesn't build the previous posts on each new "page". Instead of page 2 showing entries 21 to 40 I want it to show entries 1 - 40 and page 3 show 1 - 60 and so on. The other problem with pagination is if I get it working for my requirements I'm technically creating duplicate content over multiple urls so I don't think that is the way to go.

I guess I need to be able to somehow set a variable as to the position in the clicking order.

The only way I can get around it so far is to not paginate and use  _target="blank" on each entry url. That way the position in the archive page has been preserved.

Is there a way to resolve this or should I just stick with my work around of opening up the entries in separate browser tabs?

Any pointers would be much appreciated. 

Thanks.

**Comment by bencroker** [comment]
Your question is firmly in [htmx](https://htmx.org/) territory. Fortunately, htmx has a mechanism for interacting with the browser history API, see https://htmx.org/docs/#history

**Comment by cannon303** [comment]
> Your question is firmly in [htmx](https://htmx.org/) territory. Fortunately, htmx has a mechanism for interacting with the browser history API, see https://htmx.org/docs/#history

Thanks, I've checked that out but so far its only working in the same way as the sprig pagination functionality so I'm going around in circles a bit. I'll keep trying though.

**Comment by bencroker** [comment]
I'd suggest taking the approach of leaving out the offset and incrementing the limit each time. Rather than loading more entries, you'd be replacing them with an entirely new set, but it should then work as expected.

> NOTE: If you push a URL into the history, you must be able to navigate to that URL and get a full page back! A user could copy and paste the URL into an email, or new tab. Additionally, htmx will need the entire page when restoring history if the page is not in the history cache.

**Comment by cannon303** [comment]
I did manage to get it working using the browser history API. It worked perfectly. The page loads 4 entries and clicking the Load more button loads a further 4 entries so now 8 in total and in the address bar it says ?page=2. If you then click one of those entry links and go to that page and then click the browser back button you see ?page=2 again and 8 entries. Perfect!

Unfortunately things started to fall apart when clicking the back button again to go to page 1 and see only 4 entries. The JavaScript doesn't fire, there's no error in the console and you just get the Loading... animation that plays when you're waiting for the entries to load up.

As far as JavaScript is concerned it seems that there is a difference between a on page load and reloading the page from clicking the back button. I don't really need to be able to cycle back from say page 4 to 3 to 2 to 1 etc. It doesn't matter if that is the outcome but seeing as its causing the scripts to break it means I can't use the browser history API.

Unless there is a better solution I've decided to link those entries into a new browser tab and ditch the browser history API.
